Users/abehenderson/CSX/Closures/once.js

```diff
@@ -0,0 +1,55 @@
+// Attempt #2 (CSX Solution)
+
+function once(cb) {
+  let hasBeenCalled = false;
+  let result;
+  
+  function oncified(...args) { // Uses 'rest operator' to accept an array if needed.  
+
+    if (!hasBeenCalled) {
+      result = cb(...args); // Uses 'spread operator' to invoke 'cb' on all elements on an array, if an array is passed in (must be for an edge case, but it wasn't specified in instructions that we needed to account for an array)
+      hasBeenCalled = true;
+    }
+    return result;
+  }
+  return oncified;
+}
+
+  // Is using 'rest' and 'spread operators' considered best case scenario in a situation like this?  If this was a tech interview, I would just ask edge case questions and find out what the interviewer wants me to account for.
+
+//----------------------------------------------------------------------------------------
+// Attempt #1
+
+function once(cb) {
+  let result;
+  return function(n) {
+    if (result === undefined) { // Is there any case where using 'result' as defined or undefined may fail, vs using a boolean as in the CSX example above?  Is one preferred over the other, or is there a best practice?
+      return result = cb(n);
+    }
+      return result;
+  }
+}
+
+/* OR */
+
+function once(cb) {
+  let result;
+  return function(n) {
+    return (!result) ? result = cb(n) : result; // Is there anything wrong with using ternary operators?  I think they are more clear to read than drop down style 'if...else' statements.
```

Comment:
You're in the minority if you think ternaries are clearer than `if`/`else` blocks, but up to you. Nothing particularly wrong with them in simple examples. I'm not sure if you need the parentheses around `!result`, you should test.